src/engine/sync/abletonlink.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include <ableton/Link.hpp>
+#include <ableton/link/HostTimeFilter.hpp>
+#include <ableton/platforms/stl/Clock.hpp>
+
+#include "control/controlpushbutton.h"
+#include "engine/channels/enginechannel.h"
+#include "engine/enginebuffer.h"
+#include "engine/sync/syncable.h"
+#include "engine/sync/synccontrol.h"
+
+/// This class manages a link session.
+/// Read & update (get & set) this session for Mixxx to be a synced Link
+/// participant (bpm & phase)
+///
+/// Ableton Link Readme (lib/ableton-link/README.md)
+/// Documentation in the header (lib/ableton-link/include/ableton/Link.hpp)
+/// Ableton provides a command line tool (LinkHut) for debugging Link programs
+/// (instructions in the Readme)
+///
+/// Ableton recommends getting/setting the link session from the audio thread
+/// for maximum timing accuracy. Call the appropriate, realtime-safe functions
+/// from the audio callback to do this.
+
+// Note that the resolution of std::chrono::steady_clock is not guaranteed
+// to be high resolution, but it is guaranteed to be monotonic.
+// However, on all major platforms, it is high resolution enough.
+using AbletonLinkClockReference = ableton::platforms::stl::Clock;
```

Comment:
I think you could also use `ableton::Link::Clock` which would use a platform-specific high resolution monotonic clock (on Windows and macOS IIRC) and only fall back to the STL clock if unavailable. Or do we specifically need the STL clock?

Reply:
We use this already at other places on Mixxx, and I want to have these filtered absolute time stamps to be useable anywhere (e.g in a future VSync PR). This is why I changed this.

Reply:
Makes sense!